### DISCUSSION TITLE
Is there a global speed limit setting?

### DISCUSSION THREAD

**Discussion by AiXanadu** [curl/curl#9915] [Q&A] [answered]
### curl/libcurl version

curl 7.55.1 (Windows) libcurl/7.55.1 WinSSL
Release-Date: [unreleased]
Protocols: dict file ftp ftps http https imap imaps pop3 pop3s smtp smtps telnet tftp
Features: AsynchDNS IPv6 Largefile SSPI Kerberos SPNEGO NTLM SSL

### operating system
Windows 10 x64

In the process of using CURL, I ran into a problem.
Can I use a setting that sets a global download speed limit?
For example I have two tasks. Download speed limit is 100KB, one is 80KB and the other is 20KB?
I didn't find what I was looking for in the documentation or Api.
I only found a flag (CURLOPT_MAX_RECV_SPEED_LARGE) that sets a single session.

If what I said is no solution, then I would like to ask, can I reset the (CURLOPT_MAX_RECV_SPEED_LARGE) value for this session during the function callback?
I can set a total value myself and then adjust it dynamically.

**Comment by bagder** [comment, marked as answer]
curl has no global options at all. You can only set options on handles, and the option provided to limit transfer speeds is the one you have found. If you need to limit more than one transfer, then you need to set the option to suitable values for all involved handles.

**Comment by AiXanadu** [comment]
So can I frequently set the download speed in the CURL write/progress callback?
